src/frontend/planner_test/tests/testdata/output/index_selection.yaml

```diff
@@ -840,22 +837,20 @@
     create index idx4 on t1(b, a desc);
     select * from t1 where c = 1 and a = 2 and d = 3 order by b desc limit 10
   batch_plan: |-
-    BatchSort { order: [idx3.b DESC] }
-    └─BatchTopN { order: [idx3.a ASC, idx3.c ASC, idx3.d ASC, idx3.b DESC], limit: 10, offset: 0 }
-      └─BatchExchange { order: [], dist: Single }
-        └─BatchLimit { limit: 10, offset: 0 }
-          └─BatchScan { table: idx3, columns: [idx3.a, idx3.b, idx3.c, idx3.d], scan_ranges: [idx3.a = Int32(2) AND idx3.c = Int32(1) AND idx3.d = Int32(3)], limit: 10, distribution: UpstreamHashShard(idx3.a) }
+    BatchTopN { order: [idx3.b DESC], limit: 10, offset: 0 }
+    └─BatchExchange { order: [], dist: Single }
+      └─BatchLimit { limit: 10, offset: 0 }
+        └─BatchScan { table: idx3, columns: [idx3.a, idx3.b, idx3.c, idx3.d], scan_ranges: [idx3.a = Int32(2) AND idx3.c = Int32(1) AND idx3.d = Int32(3)], limit: 10, distribution: UpstreamHashShard(idx3.a) }
 - name: topn index selection + limit pushdown with point get + descending order point get
   sql: |
     create table t1 (a int, b int, c int);
     create index idx1 on t1(a asc, b);
     select * from t1 where a = 1 order by b limit 10;
   batch_plan: |-
-    BatchSort { order: [idx1.b ASC] }
-    └─BatchTopN { order: [idx1.a ASC, idx1.b ASC], limit: 10, offset: 0 }
-      └─BatchExchange { order: [], dist: Single }
-        └─BatchLimit { limit: 10, offset: 0 }
-          └─BatchScan { table: idx1, columns: [idx1.a, idx1.b, idx1.c], scan_ranges: [idx1.a = Int32(1)], limit: 10, distribution: UpstreamHashShard(idx1.a) }
+    BatchTopN { order: [idx1.b ASC], limit: 10, offset: 0 }
```

Comment:
The input from `BatchScan` is already sorted in the required order, and the `BatchTopN` is yet another unnecessary `BatchSort`. We can actually follow what mentioned in the issue to replace the `BatchTopN` here with `BatchLimit`.
> when turning BatchTopN into two phase physical plan, when seeing the order of input matches the order of itself, not only turning the first phase as BatchLimit, but also the second phase.

In `impl ToDistributedBatch for BatchTopN` and `impl ToLocalBatch for BatchTopN`, we didn't implement the method `to_xxx_with_order_required` specially. We can instead change to implement the `to_xxx_with_order_required`, and when seeing the required order is the same as the order of `BatchTopN`, we can turn the second phase into `BatchLimit`.